src/document.rs

```diff
@@ -314,14 +328,18 @@ impl Document {
                 .unwrap_or(default_excerpt_separator);
 
             let excerpt = if let Some(excerpt_str) = excerpt_attr {
-                excerpt_str
+                excerpt_str.to_string()
             } else if excerpt_separator.is_empty() {
-                ""
+                String::new()
```

Comment:
I tried with a number of different options in each clause, but I am not fluent enough with the vagaries of the rust type/borrowing/lifetime system, so I'll go for the simple fix.  Instead of allocating a new string, I'll just convert the `""` to a string using `"".to_string()`.